gene2phenotype_project/gene2phenotype_app/serializers/user.py

```diff
@@ -213,6 +229,49 @@ class Meta:
                         }
                         }
         
+class AddUserToPanelSerializer(serializers.ModelSerializer):
+    user = serializers.CharField(write_only=True)
+    panel = serializers.ListField(child=serializers.CharField(), allow_empty=False, write_only=True)
+
+    def validate(self, attrs):
+        user = attrs.get('user', '')
+        if user is None:
+            raise serializers.ValidationError({'message': 'user email is required to add user to panel'})
+        if user:
+            if not User.objects.filter(email=user).exists():
+                raise serializers.ValidationError({'message': "User does not exist, please create a user and add user to panel"})
+        
+        panels = attrs.get('panel', [])
+        if panels:
+            for panel in panels:
+                if not Panel.objects.filter(name=panel).exists():
+                    raise serializers.ValidationError({'message': "Panel does not exists, Please check panel"})
+        
+        return attrs
+    
+
+    def create(self, validated_data):
+        user_email = validated_data.get('user')
+        panels = validated_data.get('panel')
+
+        user_obj = User.objects.get(email=user_email)
+        for panel in panels:
+            panel_obj = Panel.objects.get(name=panel)
+
+            if UserPanel.objects.filter(user=user_obj, panel=panel_obj, is_deleted=1).exists():
+                raise serializers.ValidationError({'message': f"User {user_email} used to exists in this {panel}, Update instead"})
+
+            if not UserPanel.objects.filter(user=user_obj, panel=panel_obj, is_deleted=0).exists():
+                user_panel = UserPanel.objects.create(user=user_obj, panel=panel_obj, is_deleted=0)
+            raise serializers.ValidationError({'message': f"User {user_email} already exists in this {panel}"})
```

Comment:
The `else` is missing.
Right now the endpoint always returns this error message.